pkg/kotsutil/kots.go

```diff
@@ -564,6 +585,53 @@ func LoadKotsKindsFromPath(fromDir string) (*KotsKinds, error) {
 	return &kotsKinds, nil
 }
 
+func KotsKindsFromMap(kotsKindsMap map[string][]byte) (*KotsKinds, error) {
+	kotsKinds := EmptyKotsKinds()
+
+	for path, content := range kotsKindsMap {
+		if err := kotsKinds.addKotsKind(content); err != nil {
+			return nil, errors.Wrapf(err, "failed to parse kots kind %s", path)
+		}
+	}
+
+	return &kotsKinds, nil
+}
+
+func WriteKotsKinds(kotsKinds map[string][]byte, rootDir string, overwrite bool) error {
+	_, err := os.Stat(rootDir)
+	if err == nil {
+		if overwrite {
```

Comment:
this should always be true. there shouldn't be an option to _not_ overwrite. always cleanup/remove the existing directory.